src/main/java/utils/command/DeleteTagCommand.java

```diff
@@ -17,20 +19,33 @@ public DeleteTagCommand(String tagName) {
         this.tagName = tagName;
     }
 
-    private void removeTagFromCards(CardList cardList, TagList tagList, UserInterface ui) {
+    /**
+     * Remoove the tagName tag from the tagList and all the cards that currently have the tagName tag.
+     *
+     * @param cardList The cardList from which to look for the cards with tagName tag.
+     * @param tagList  The tagList from which to delete the tagName tag.
+     * @param ui       The userInterface to print the success of removal of the tag from the cards.
+     * @throws InkaException
+     */
+    private void removeTagFromCards(CardList cardList, TagList tagList, UserInterface ui) throws InkaException {
         Tag tag = tagList.findTagFromName(tagName);
+        if (tag == null) {
+            throw new TagNotFoundException();
+        }
         tagUUID = tag.getUUID();
 
         //for each card whose uuid is listed under the tag, remove the tag uuid from that card
         for (CardUUID cardUUID : tag.getCardsUUID()) {
             Card affectedCard = cardList.findCardFromUUID(cardUUID);
+            assert affectedCard != null;
```

Comment:
^ Same issue as above

Reply:
Yeap same exception checking has been done in `cardList.findCardFromUUID(cardUUID)`